File: easy/1732-find-the-highest-altitude/solution.py
from typing import List


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        """
        1732. Find the Highest Altitude
        Time: O(n)
        Space: O(1)
        """
        altitude = 0
        highest = 0

        for g in gain:
            altitude += g
            highest = max(highest, altitude)

        return highest

File: easy/1732-find-the-highest-altitude/solution.java
/**
 * 1732. Find the Highest Altitude
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int largestAltitude(int[] gain) {
        int altitude = 0;
        int highest = 0;

        for (int g : gain) {
            altitude += g;
            highest = Math.max(highest, altitude);
        }

        return highest;
    }
}
